apps/admin_app/assets/js/views/product/edit.js

```diff
@@ -84,74 +84,74 @@ function deleteError(data) {
   console.log('errror data', data)
 }
 
-function setup_product(){
-    let previousOptionValue = ""
+function setup_product() {
+  let previousOptionValue = ""
 
-    // This handle the variation theme selection
-    $('#product_theme_id').on('change', function (e) {
-      $('#theme_change_modal').modal(`show`);
-    })
+  // This handle the variation theme selection
+  $('#product_theme_id').on('change', function (e) {
+    $('#theme_change_modal').modal(`show`);
+  })
 
-    $("#theme_change_confirm").click(function(e){
-        var optionSelected = $('#product_theme_id');
-        var valueSelected = optionSelected.val();
-        const product_id = $(this)
-          .parents()
-          .find('#product_id');
-        const new_variant = $(this)
-          .parents()
-          .find('#new_variant');
-        var link = "/products/" + product_id.val() + "/variant/new?theme_id=" + valueSelected
-        new_variant.attr("href", link)
-
-        get_variation_options(valueSelected, product_id.val())
-    })
+  $("#theme_change_confirm").click(function (e) {
+    var optionSelected = $('#product_theme_id');
+    var valueSelected = optionSelected.val();
+    const product_id = $(this)
+      .parents()
+      .find('#product_id');
+    const new_variant = $(this)
+      .parents()
+      .find('#new_variant');
+    var link = "/products/" + product_id.val() + "/variant/new?theme_id=" + valueSelected
+    new_variant.attr("href", link)
+
+    get_variation_options(valueSelected, product_id.val())
+  })
 
-    $(".option-value")
-    .blur(function(evt){
+  $(".option-value")
+    .blur(function (evt) {
       var td = $(evt.target)
       let option_value_id = td.data("option_value_id")
       let value = td[0].innerText
-      if(value !== "")
+      if (value !== "")
         update_option_value(option_value_id, value)
       else {
         td.text(previousOptionValue)
       }
     })
 
-    $(".option-value")
-    .focusin(function(evt){
+  $(".option-value")
+    .focusin(function (evt) {
       var td = $(evt.target)
       previousOptionValue = td[0].innerText
     })
 
-    var product_id = $("#product_id").val();
-    var theme_id = $("#product_theme_id").val();
-    if(theme_id !== undefined && !(theme_id == ""))
-      get_variation_options(theme_id, product_id)
+  var product_id = $("#product_id").val();
+  var theme_id = $("#product_theme_id").val();
+  if (theme_id !== undefined && !(theme_id == ""))
+    get_variation_options(theme_id, product_id)
 }
 
-function update_option_value(id, value){
-    $.ajax({
-      type: "POST",
-      url: 'http://localhost:3000/api/v1/product_option_values/' + id,
-      crossDomain: true,
-      data: {value: value},
-      success: function(response){
-      }
-    })
-  }
+function update_option_value(id, value) {
+  $.ajax({
+    type: "POST",
+    url: 'http://localhost:3000/api/v1/product_option_values/' + id,
```

Comment:
@SagarKarwande @jyotigautam @jaypal412 this is outright wrong. `http://localhost:3000` can never be used like this imo always picked from env vars.

Reply:
@ashish173 Config for env is not setup so the endpoint are used directly.

Reply:
we would not need to call `snitch API` here, we should create an endpoint in `admin_app` for the same. Both are independent apps. Admin might need special DB scopes in future like permissions to fetch data intended for admin only. And api could have different scope, like get me data that scoped by normal users.

Reply:
@SagarKarwande @ashish173 we can also use route proxy, like forwarding the route for one app to another. check this https://elixirforum.com/t/umbrella-app-routing-with-plug-router-app-phoenix-app/10213